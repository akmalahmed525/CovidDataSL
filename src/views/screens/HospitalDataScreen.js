// @flow
import React, {Component} from 'react';
import {View, TouchableOpacity, Text, FlatList, StyleSheet} from 'react-native';
import {connect} from 'react-redux';
import {human} from 'react-native-typography';

import {t} from 'i18n';
import {normalize} from 'utils';
import themes from 'themes';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: themes.colors.white
  },
  banner: {
    alignItems: 'center',
    justifyContent: 'center',
    padding: normalize(10)
  },
  text: {
    textAlign: 'center',
    fontFamily: 'Rubik-Regular'
  },
  cardTitle: {
    textAlign: 'left',
    fontFamily: 'Rubik-Regular'
  },
  cardSubTitle: {
    textAlign: 'left',
    fontFamily: 'Rubik-Light'
  },
  card: {
    flexDirection: 'column',
    shadowColor: themes.colors.black,
    shadowOffset: {
      width: 0,
      height: 2
    },
    shadowOpacity: themes.dimens.cardShadowOpacity,
    shadowRadius: themes.dimens.cardShadowRadius,
    elevation: themes.dimens.cardElevation,
    backgroundColor: themes.colors.white,
    padding: themes.dimens.cardPadding,
    margin: themes.dimens.cardMargin,
    borderRadius: themes.dimens.cardRadius
  }
});

type HospitalItemProps = {
  name: string,
  localPatients: any,
  foreignPatients: any,
  cumulativePatients: any
};
const HospitalDataItem = ({
  name,
  localPatients,
  foreignPatients,
  cumulativePatients
}: HospitalItemProps) => (
  <TouchableOpacity style={styles.card} activeOpacity={0.85}>
    <Text style={[human.body, styles.cardTitle]}>{name}</Text>
    <Text style={[human.footnote, styles.cardSubTitle]}>{localPatients}</Text>
    <Text style={[human.footnote, styles.cardSubTitle]}>{foreignPatients}</Text>
    <Text style={[human.footnote, styles.cardSubTitle]}>
      {cumulativePatients}
    </Text>
  </TouchableOpacity>
);

type Props = {
  locale: 'en',
  data: Object
};
class HospitalDataScreenView extends Component<Props> {
  render() {
    const {locale, data} = this.props;
    const hospitalData = data.hospital_data;
    return (
      <View style={styles.container}>
        <View style={styles.banner}>
          <Text style={[human.title1, styles.text]}>
            {t(`placeholders.hospital_data`, locale)}
          </Text>
        </View>
        <FlatList
          renderItem={({item}) => (
            <HospitalDataItem
              name={item.hospital.name}
              localPatients={t(
                `placeholders.local_patients`,
                locale,
                item.cumulative_local
              )}
              foreignPatients={t(
                `placeholders.foreign_patients`,
                locale,
                item.cumulative_foreign
              )}
              cumulativePatients={t(
                `placeholders.cumulative_total`,
                locale,
                item.cumulative_total
              )}
            />
          )}
          data={hospitalData}
          keyExtractor={(item) => `${item.id}`}
        />
      </View>
    );
  }
}

const mapStateToProps = (state) => ({
  locale: state.app.state,
  data: state.app.data
});

const HospitalDataScreen = connect<Props, *, *, *, *, *>(
  mapStateToProps,
  {}
)(HospitalDataScreenView);
export default HospitalDataScreen;
