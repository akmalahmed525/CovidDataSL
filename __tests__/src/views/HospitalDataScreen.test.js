import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import HospitalDataScreen from 'src/views/screens/HospitalDataScreen';

const mockStore = configureStore([]);

describe('Hospital data screen', () => {
  let store;
  let hospitalDataScreen;

  const data = {
    hospital_data: [
      {
        id: 1,
        hospital_id: 1,
        cumulative_local: 943,
        cumulative_foreign: 75,
        treatment_local: 2,
        treatment_foreign: 0,
        created_at: '2020-06-04 13:49:42',
        updated_at: null,
        deleted_at: null,
        cumulative_total: 1018,
        treatment_total: 2,
        hospital: {
          id: 1,
          name: 'National Institute of Infectious Diseases',
          name_si:
            '\u0db6\u0ddd\u0dc0\u0db1 \u0dbb\u0ddd\u0d9c \u0db4\u0dd2\u0dc5\u0dd2\u0db6\u0db3 \u0da2\u0dcf\u0dad\u0dd2\u0d9a \u0d86\u0dba\u0dad\u0db1\u0dba',
          name_ta:
            '\u0ba4\u0bc7\u0b9a\u0bbf\u0baf \u0ba4\u0bca\u0bb1\u0bcd\u0bb1\u0bc1 \u0ba8\u0bcb\u0baf\u0bbf\u0baf\u0bb2\u0bcd \u0ba8\u0bbf\u0bb1\u0bc1\u0bb5\u0ba9\u0bae\u0bcd',
          created_at: '2020-04-11 08:45:06',
          updated_at: '2020-04-11 08:45:06',
          deleted_at: null
        }
      }
    ]
  };

  const locale = 'en';

  const props = {data, locale};

  beforeEach(() => {
    store = mockStore({app: props});
    hospitalDataScreen = renderer.create(
      <Provider store={store}>
        <HospitalDataScreen {...props} />
      </Provider>
    );
  });

  test('Must match the snapshot', () => {
    expect(hospitalDataScreen.toJSON()).toMatchSnapshot();
  });
});
