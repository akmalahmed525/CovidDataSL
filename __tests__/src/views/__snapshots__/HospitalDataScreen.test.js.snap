// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hospital data screen Must match the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 24,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#000000",
            "fontFamily": "System",
            "fontSize": 28,
            "fontWeight": "400",
            "letterSpacing": 0.355469,
            "lineHeight": 34,
          },
          Object {
            "fontFamily": "Rubik-Regular",
            "textAlign": "center",
          },
        ]
      }
    >
      Covid-19 treating hospital's
    </Text>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "created_at": "2020-06-04 13:49:42",
          "cumulative_foreign": 75,
          "cumulative_local": 943,
          "cumulative_total": 1018,
          "deleted_at": null,
          "hospital": Object {
            "created_at": "2020-04-11 08:45:06",
            "deleted_at": null,
            "id": 1,
            "name": "National Institute of Infectious Diseases",
            "name_si": "බෝවන රෝග පිළිබඳ ජාතික ආයතනය",
            "name_ta": "தேசிய தொற்று நோயியல் நிறுவனம்",
            "updated_at": "2020-04-11 08:45:06",
          },
          "hospital_id": 1,
          "id": 1,
          "treatment_foreign": 0,
          "treatment_local": 2,
          "treatment_total": 2,
          "updated_at": null,
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 24,
              "elevation": 5,
              "flexDirection": "column",
              "margin": 24,
              "opacity": 1,
              "padding": 24,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 17,
                  "fontWeight": "400",
                  "letterSpacing": -0.408,
                  "lineHeight": 22,
                },
                Object {
                  "fontFamily": "Rubik-Regular",
                  "textAlign": "left",
                },
              ]
            }
          >
            National Institute of Infectious Diseases
          </Text>
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 13,
                  "fontWeight": "400",
                  "letterSpacing": -0.078,
                  "lineHeight": 18,
                },
                Object {
                  "fontFamily": "Rubik-Light",
                  "textAlign": "left",
                },
              ]
            }
          >
            Total number of local patients 943
          </Text>
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 13,
                  "fontWeight": "400",
                  "letterSpacing": -0.078,
                  "lineHeight": 18,
                },
                Object {
                  "fontFamily": "Rubik-Light",
                  "textAlign": "left",
                },
              ]
            }
          >
            Total number of foreign patients 75
          </Text>
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 13,
                  "fontWeight": "400",
                  "letterSpacing": -0.078,
                  "lineHeight": 18,
                },
                Object {
                  "fontFamily": "Rubik-Light",
                  "textAlign": "left",
                },
              ]
            }
          >
            Cumulative number of patients 1018
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
